File: src/main/java/pr/eleks/we_at_her/services/api/ApiService.java
package pr.eleks.we_at_her.services.api;

import pr.eleks.we_at_her.dto.WeatherSampleDto;
import pr.eleks.we_at_her.exceptions.PropertyNotFoundException;
import pr.eleks.we_at_her.exceptions.WrongApiResponseException;

public interface ApiService {
    WeatherSampleDto getWeatherSampleFromApi(String latitude, String longitude, String lang, String units) throws PropertyNotFoundException, WrongApiResponseException;

    String getName();

    void prepareParameters(String latitude, String longitude, String lang, String units) throws PropertyNotFoundException;

    void prepareBaseUrl() throws PropertyNotFoundException;
}

File: src/main/kotlin/pr/eleks/we_at_her/services/api/ApiService.kt
package pr.eleks.we_at_her.services.api;

import pr.eleks.we_at_her.dto.weather.WeatherSampleDto;
import pr.eleks.we_at_her.exceptions.PropertyNotFoundException;
import pr.eleks.we_at_her.exceptions.WrongApiResponseException;

interface ApiService {
    fun getWeatherSampleFromApi(latitude: String?, longitude: String?, lang: String?, units: String?): WeatherSampleDto
    fun getName(): String
    fun prepareParameters(latitude: String?, longitude: String?, lang: String?, units: String?)
    fun prepareBaseUrl()
}
